### ISSUE TITLE
Allow for a `description` to be attached to an API Response Example

### ISSUE THREAD

**Issue by r-colvin** [PaloAltoNetworks/docusaurus-openapi-docs#283]
### Is your feature request related to a problem?
- No

It would be hugely beneficial to be able to provide users with a description of what a response means, especially in the case of an error. It should be one description per example

### Describe the solution you'd like

a spec such as
```
content:
   application/json:
     examples:
         ABC
              description: a description of this response
               value:
                   key: value
```

with a result such as 

![image](https://user-images.githubusercontent.com/18327771/192559372-d7fb37c3-3feb-4be5-8a4b-ff108cd2602a.png)

**Comment by sserrata** [comment]
I believe that OpenAPI already provides a `summary` property for this purpose: https://swagger.io/docs/specification/adding-examples/

**Comment by r-colvin** [comment]
@sserrata in my testing it works great for requests, but not responses. The description is ignored. I may have messed up the syntax - I can put a test together and check again

I’ve read the attached link several times - unless I have missed something, it mostly covers examples in requests?

**Comment by r-colvin** [comment]
@

**Comment by r-colvin** [comment]
Fat fingers! Only meant to comment not close

**Comment by sserrata** [comment]
Sorry, what I meant to say is that we likely want to check to see if the example includes a `summary` property and render that above the example if it exists.

**Comment by sserrata** [comment]
For example, something like this:

```yaml
      responses:
        "200":
          description: successful operation
          headers:
            X-Rate-Limit:
              description: calls per hour allowed by the user
              schema:
                type: integer
                format: int32
            X-Expires-After:
              description: date in UTC when token expires
              schema:
                type: string
                format: date-time
          content:
            application/json:
              schema:
                type: string
              examples:
                response:
                  summary: OK response in JSON format
                  value: OK
            application/xml:
              schema:
                type: string
              examples:
                response:
                  summary: OK response in XML format
                  value: <Message> OK </Message>
            text/plain:
              examples:
                response:
                  summary: OK response in plain text format
                  value: OK
```

Should result in this:

https://user-images.githubusercontent.com/9343811/192628054-73f1fed1-579f-4fe3-bfe7-f2b2ce385869.mov

**Comment by sserrata** [comment]
If that works, I can open a PR to introduce this functionality.

**Comment by r-colvin** [comment]
@sserrata 

looks good

is the schema line optional? can i have this for example:
![image](https://user-images.githubusercontent.com/18327771/192631223-fac05c2e-46b9-43e5-a1d1-70597fd35c3d.png)

to produce something like this

![image](https://user-images.githubusercontent.com/18327771/192632331-a5d50cd1-103c-45ff-9ab4-6c6577b41928.png)

**Comment by sserrata** [comment]
What you're describing is technically possible but I believe it might violate best practices. Essentially, you'd be defining a response with no true schema. Examples, IMO, are not a substitute for a schema.

**Comment by r-colvin** [comment]
@sserrata  yeah, thats OK; I think i can live with that :)

**Comment by sserrata** [comment]
Support for this feature included in v1.2.2
